hip-associate-dissociate.md

```diff
@@ -0,0 +1,127 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
+title: Associate and Dissociate Tokens via Facade Contract
+author: Luke Lee <@lukelee-sl>
+working-group: Nana Essilfie-Conduah, Danno Ferrin <@shemnon> <@nana-ec>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+created: 2023-04-19
+discussions-to: 
+updated: 
+requires: 
+replaces: 
+superseded-by: 
+---
+
+## Abstract
+
+[HIP-218](https://hips.hedera.com/hip/hip-218) proposed a way to enable treating HTS tokens in an equivalent manner
+to ERC-20 and ERC-721 tokens by creating a proxy redirect facade contract.  This proposal extends
+this functionality to include the HTS `associate` and `dissociate` functions.
+
+## Motivation
+
+There are some functions that are unique to Hedera HTS tokens that are not part of the ERC standard.
+A mechanism should be provided to enable developers to call these functions on HTS tokens in a way that is familiar
+to users of ERC-20 and ERC-721 tokens.
+
+## Rationale
+
+By adding `associate` and `dissociate` functions to the token proxy redirect facade contract, functionality
+that is only available to HTS tokens can be included to the available capabilities of token contracts.
+
+## User stories
+
+1. As a developer, I want to be able to call the `associate` and `dissociate` functions on HTS tokens in a way that is similar to calling functions on the IERC20 and IERC721 interfaces.
```

Comment:
As a DApp I want to support a users ability to execute the association flows without having to call an intermediate contract.
Something along that line

Reply:
and it's really great UX to be able to include the token associate in the same contract call as a claim function, altho not sure if that's gonna be possible in any way under the new security model. Having to associate in an tx and then claim in another tx is meh UX. 

Granted, the same thing could be said about requiring spending approval in a separate tx (just pointing out some potential hypocrisy on my side). Guess I see the associate action is just so much lower risk than a spending approval, so perhaps it doesn't require a separate tx.